src/sentry/explore/endpoints/explore_saved_query_starred.py

```diff
@@ -0,0 +1,72 @@
+from rest_framework import serializers, status
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry import features
+from sentry.api.api_owners import ApiOwner
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import region_silo_endpoint
+from sentry.api.bases.organization import OrganizationEndpoint, OrganizationPermission
+from sentry.explore.models import ExploreSavedQuery, ExploreSavedQueryStarred
+from sentry.models.organization import Organization
+
+
+class StarQuerySerializer(serializers.Serializer):
+    starred = serializers.BooleanField(required=True)
+    position = serializers.IntegerField(required=False)
+
+    def validate(self, data):
+        if not data["starred"] and "position" in data:
+            raise serializers.ValidationError("Position is only allowed when starring a query.")
+        return data
+
+
+class MemberPermission(OrganizationPermission):
+    scope_map = {
+        "POST": ["member:read", "member:write"],
+    }
+
+
+@region_silo_endpoint
+class ExploreSavedQueryStarredEndpoint(OrganizationEndpoint):
+    publish_status = {
+        "POST": ApiPublishStatus.EXPERIMENTAL,
+    }
+    owner = ApiOwner.PERFORMANCE
+    permission_classes = (MemberPermission,)
+
+    def has_feature(self, organization, request):
+        return features.has(
+            "organizations:performance-trace-explorer", organization, actor=request.user
+        )
+
+    def post(self, request: Request, organization: Organization, id: int) -> Response:
+        """
+        Update the starred status of a saved Explore query for the current organization member.
+        """
+        if not self.has_feature(organization, request):
+            return self.respond(status=404)
+
+        serializer = StarQuerySerializer(data=request.data)
+        if not serializer.is_valid():
+            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+        is_starred = serializer.validated_data["starred"]
+
+        try:
+            query = ExploreSavedQuery.objects.get(id=id, organization=organization)
```

Comment:
Should this restrict to the current user? Right now I could provide an ID to any saved query in the organization, and get a 20X response code, which doesn't seem right.

Reply:
We haven't finalized how permissions should look like yet for saved queries, and it's not part of the initial release milestone. At the moment we are just allowing all access to all saved queries.

There will be a follow up in the next milestone to add permissions to saved queries which may add restrictions to what queries a user can star.